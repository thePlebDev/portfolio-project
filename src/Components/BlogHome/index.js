import React,{useState,useEffect} from 'react';
import styled from 'styled-components';

import axiosUtil from '../../utils/AxiosUtil'


import ArticleCard from '../ArticleCard';
import useApiCall from '../../Hooks/useApiCall'


const  Container= styled.div`
  position:relative;
`


const H1 = styled.h1`
  opacity:0.7;
  text-align:center;
  position:relative;
  ::after{
    content:"";
    background-color: red;
    height:2px;
    width:50px;
    bottom:0;
    left:0;
    right:0;
    margin:0 auto;
    position:absolute;
  }
`

const ArticleContainer = styled.div`
    display:grid;
    grid-template-columns:repeat(auto-fit,minmax(350px,1fr));
    grid-gap:2rem;
`

const BlogHome =({apiHook=useApiCall})=>{
  const {state,setState} = apiHook(axiosUtil.get,'/v1/blog/all')
  //const [blogState,setBlogState] = useState(state)

  // useEffect(()=>{
  //   if(state){
  //     sessionStorage.setItem('posts',JSON.stringify(state.posts))
  //   }
  // },[state])

  return(
    <Container>
      <H1>Articles</H1>
      <ArticleContainer data-testid="container">
        {
          state
            ?
          state.posts.map((item,index)=>{
            return <ArticleCard key={item._id} id={item._id} setBlogState={setState} blogState={state} key={index} image={item.image} filters={item.filters} title={item.title} description={item.description} />
          })
          :
          ''
        }
      </ArticleContainer>
    </Container>
  )
}


export default BlogHome
