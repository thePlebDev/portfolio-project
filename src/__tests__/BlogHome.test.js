import React from 'react'

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import BlogHome from '../Components/BlogHome'

//1) test the snapshot
//2) test the props
//3) test the events


describe('Testing the BlogHome component',()=>{
  let wrapper;
  const mockApiHook = jest.fn(()=>{
    let state ={
      posts:[{name:'it do be like that sometimes'},
      {name:'it do be like that sometimes'},{name:'it do be like that sometimes'},
      {name:'it do be like that sometimes'},
      {name:'it do be like that sometimes'},{name:'it do be like that sometimes'}]
    }
    let setState = ()=>{}
      return{
        state,
        setState
    }
  })
  beforeEach(()=>{
    wrapper = shallow(<BlogHome apiHook={mockApiHook} state={'another one'}/>)
  })
  it('should test the snapshot',()=>{
    const tree = renderer.create(<BlogHome />).toJSON
    expect(tree).toMatchSnapshot();
  })
  it('should render all the parts properly',()=>{
    let me = shallow(<BlogHome apiHook={mockApiHook} state={'another one'}/>)
    // the function is being called 3 times, why?
    expect(mockApiHook.mock.calls.length).toBe(3)


  })
})
