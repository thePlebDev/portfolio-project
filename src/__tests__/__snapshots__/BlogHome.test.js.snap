// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the BlogHome component should test the snapshot 1`] = `[Function]`;
